<launch>
  <arg name="namespace" default="stixel_world"/>
  
  <arg name="useGraph" default="true" />
  <arg name="useCostMatrix" default="false" />
  <arg name="useObjects" default="true" />
  <arg name="twoLevelsTracking" default="true" />
  
  <arg name="SADFactor" default="0.0" />
  <arg name="heightFactor" default="0.0" />
  <arg name="polarDistFactor" default="0.0" />
  <arg name="polarSADFactor" default="0.0" />
  <arg name="histBatFactor" default="1.0" />
  <arg name="increment" default="1" />
  <param name="use_sim_time" value="true" />

   <node pkg="tf" type="static_transform_publisher" name="map2odom" args="0 0 0 1.57 0 0 map odom 100" />
    <node pkg="tf" type="static_transform_publisher" name="odom2basefootprint" args="0 0 0 0 0 0 odom base_footprint 100" />
    <node pkg="tf" type="static_transform_publisher" name="basefootprint2baselink" args="0 0 0 0 0 0 base_footprint base_link 100" />
    <node pkg="tf" type="static_transform_publisher" name="baselink2leftcam" args="1.0 0 0 -1.57 0 0 base_link left_cam 100" />
<!--     <node pkg="tf" type="static_transform_publisher" name="map2leftcam" args="0 0 0 0 0 0 map left_cam 100" /> -->

    <!-- <node pkg="tf" type="static_transform_publisher" name="odom2baselink" args="0 0 0 0 0 0 odom base_link 100" /> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="baselink2leftcam" args="0 0 0 0 0 0 base_link left_cam 100" /> -->

<!--    <node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer" output="screen">
        <remap from="stereo" to="$(arg namespace)/stixels_world"/>
        <param name="base_link_frame_id" value="base_footprint" />
        <param name="odom_frame_id" value="odom" />
        
    </node>-->
    <!--     <remap from="image" to="image_rect"/> -->

    <group ns="$(arg namespace)">
    <!--     <node launch-prefix="gdb -ex run &#45;&#45;args" name="stixels_world" pkg="stixel_world" type="stixels_world_node" output="screen" required="true" > -->
        <node name="stixels_world" pkg="stixel_world" type="stixels_world_node" output="screen" required="true" >
            <param name="useGraph" value="$(arg useGraph)" />
            <param name="useCostMatrix" value="$(arg useCostMatrix)" />
            <param name="useObjects" value="$(arg useObjects)" />

            <param name="SADFactor" value="$(arg SADFactor)" />
            <param name="heightFactor" value="$(arg heightFactor)" />
            <param name="polarDistFactor" value="$(arg polarDistFactor)" />
            <param name="polarSADFactor" value="$(arg polarSADFactor)" />
            <param name="histBatFactor" value="$(arg histBatFactor)" />
            <param name="twoLevelsTracking" value="$(arg twoLevelsTracking)" />
            <param name="increment" value="$(arg increment)" />

            <remap from="~/pointCloudStixels" 
                   to="/$(arg namespace)/pointCloudStixels" />
            <remap from="~/fakePointCloud" 
                to="/$(arg namespace)/clearing_point_cloud" />
        </node>
    </group>

    <node launch-prefix="optirun" pkg="rviz" type="rviz" name="rviz" required="false"  args="&#45;&#45;display-config 
      $(find stixel_world)/config/stixels_world.rviz"/>
    
</launch>
